app/controllers/ems_common.rb

```diff
@@ -427,27 +427,66 @@ def update_button_reset
 
   def update_button_validate
     verify_ems = find_by_id_filtered(model, params[:id])
-    validate_credentials verify_ems
+    queue_credential_validation_task verify_ems
   end
   private :update_button_validate
 
-  def validate_credentials(verify_ems)
+  def queue_credential_validation_task(verify_ems)
     set_record_vars(verify_ems, :validate)
+    creds = set_record_creds(verify_ems)
+
     @in_a_form = true
     @changed = session[:changed]
 
     # validate button should say "revalidate" if the form is unchanged
     revalidating = !edit_changed?
-    result, details = verify_ems.authentication_check(params[:type], :save => revalidating)
-    if result
-      add_flash(_("Credential validation was successful"))
+
+    args = {
+      :ems        => verify_ems,
+      :creds      => creds,
```

Comment:
> why we can't save the authentication temporarily and roll it back if it's not what we want.

Well, if you're changing the auth on an existing record, you'd have to find a way to not lose the original values.

I personally think the easiest way to do this overall is to

1. Have a class method on each method that can verify raw credentials.  It should be able to use `raw_connect`, and thus no EMS object is needed.
2. Change the UI to not use a temporary EMS for validation.  It can just put the expected info on the queue for the class method, which is likely just endpoint info, user, pass, and maybe some helper options like verify_ssl.
3. Change the instance method for verifying credentials to just call the class method.  That can probably still be used for re-verifying the credentials of an existing EMS.